script.js

```diff
@@ -0,0 +1,206 @@
+modal.appendChild(modalheading);
+modalheading.classList.add('modalheading');
+
+const headingdetailssec = document.createElement('div');
+modal.appendChild(headingdetailssec);
+headingdetailssec.classList.add('headingdetailssec');
+
+const headingdetails = document.createElement('div');
+headingdetailssec.appendChild(headingdetails);
+headingdetails.classList.add('headingdetails');
+
+const headingdetails1 = document.createElement('div');
+headingdetailssec.appendChild(headingdetails1);
+headingdetails1.classList.add('headingdetails1');
+
+const headingdate = document.createElement('div');
+headingdetailssec.appendChild(headingdate);
+headingdate.classList.add('headingdate');
+
+const modalimage = document.createElement('div');
+modal.appendChild(modalimage);
+modalimage.classList.add('modalimage');
+
+const modalparagraph = document.createElement('div');
+modal.appendChild(modalparagraph);
+modalparagraph.classList.add('modalparagraph');
+
+const paragraphtext = document.createElement('p');
+modalparagraph.appendChild(paragraphtext);
+paragraphtext.classList.add('paragraphtext');
+
+const tech = document.createElement('div');
+modal.appendChild(tech);
+tech.classList.add('tech');
+
+const modalbutton = document.createElement('div');
+modal.appendChild(modalbutton);
+modalbutton.classList.add('modalbutton');
+
+const openmodal = document.querySelector('.modal');
+const overlay = document.querySelector('.overlay');
+
+for (let i = 0; i < pdetails.length; i += 1) {
+  button[i].addEventListener('click', () => {
+    openmodal.classList.add('active');
+    overlay.classList.add('active');
+
+    let tech1 = '';
+    pdetails[i].ptech.forEach((ptech) => {
+      tech1 += `<li>${ptech}</li>`;
+    });
+
+    modalheading.innerHTML += `${pdetails[i].pname}<button class='modal-close'>&times;</button>`;
+    headingdetails.innerHTML += `${pdetails[i].pnamedetails}`;
+    headingdetails1.innerHTML += `<div class='round1'> </div> ${pdetails[i].pnamedetails1} `;
+    headingdate.innerHTML += `<div class='round2'> </div> ${pdetails[i].pnamedate}`;
+    modalimage.innerHTML += `<img class = 'image' src= '${pdetails[i].pimage}' alt='Image'>`;
+    paragraphtext.innerHTML += `${pdetails[i].pdescription}`;
+    tech.innerHTML += `<ul>${tech1}</ul>`;
+
+    modalbutton.innerHTML += `
+   <a href = '${pdetails[i].plive}' <button class = 'seelive-button-${i}' >See Live <img class = 'exportimage' src='./img/export.svg' alt = 'export image'> </button></a>
+   <a href = '${pdetails[i].psource}'<button class = 'seesource-button-${i}' >See Source <img class = 'githubimage' src='./img/github_blue.svg' alt = 'git hub image'> </button></a>`;
+  });
+}
+
+const closemodalbuttons = document.querySelector('.modalheading');
+
+closemodalbuttons.addEventListener('click', () => {
+  openmodal.classList.remove('active');
+  overlay.classList.remove('active');
+  // openmodal.style.display = 'none';
+  // overlay.style.display = 'none';
```

Comment:
- [ ] Commenting out code instead of properly removing it is considered a bad practice in software development as it can make the code base cluttered and difficult to maintain. The commented-out code can also lead to confusion and misinterpretation for other developers who are working on the same codebase.

- It's better to remove the code completely if it's no longer being used or to add a proper comment explaining the purpose of the code if it needs to be kept for reference.